File: Data Structures & Algorithms/binary-search/submission-5.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        if len(nums) == 1:
            if nums[0] == target:
                return 0
            else:
                return -1
        

        l = 0
        r = len(nums) - 1
        m = (r // 2) + 1

        print(l)
        print(m)
        print(r)

        while l <= r:
            if target == nums[m]:
                return m
            elif nums[m] < target:
                l = m + 1
                m = (l + r) // 2
            elif nums[m] > target:
                r = m - 1
                m = (l + r) // 2
        return -1
        
        

File: Data Structures & Algorithms/two-integer-sum-ii/submission-5.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        Map<Integer, Integer> mp = new HashMap<>();

        for (int i = 0; i < numbers.length; i++) {
            int complement = target - numbers[i];
            if (mp.containsKey(complement)) {
                return new int[] {mp.get(complement) + 1, i + 1};
            } else {
                mp.put(numbers[i], i);
            }
        }
        return new int[] {-1, -1};
    }
}
